src/Internal/Serialization/ToBytes.purs

```diff
@@ -17,32 +28,47 @@ import Ctl.Internal.Serialization.Types
   , TransactionOutput
   , TransactionUnspentOutput
   , TransactionWitnessSet
-  , VRFKeyHash
   , Value
   )
 import Ctl.Internal.Types.ByteArray (ByteArray)
+import Ctl.Internal.Types.CborBytes (CborBytes(CborBytes))
+import Untagged.Castable (class Castable)
 import Untagged.Union (type (|+|))
 
+type SerializationData = Address
+  |+| AuxiliaryDataHash
+  |+| ByronAddress
+  |+| DataHash
+  |+| Ed25519KeyHash
+  |+| Ed25519Signature
+  |+| GenesisDelegateHash
+  |+| GenesisHash
+  |+| Mint
+  |+| NativeScript
+  |+| PlutusData
+  |+| PoolMetadataHash
+  |+| Redeemers
+  |+| ScriptDataHash
+  |+| ScriptHash
+  |+| StakeCredential
+  |+| Transaction
+  |+| TransactionBody
+  |+| TransactionHash
+  |+| TransactionOutput
+  |+| TransactionUnspentOutput
+  |+| TransactionWitnessSet
+  |+| Value
+  |+| VRFKeyHash
+
+-- and more as needed
+
 -- NOTE returns cbor encoding for all but hash types, for which it returns raw bytes
-foreign import toBytes
-  :: ( Transaction
-         |+| TransactionBody
-         |+| TransactionOutput
-         |+| TransactionUnspentOutput
-         |+| TransactionHash
-         |+| DataHash
-         |+| PlutusData
-         |+| TransactionWitnessSet
-         |+| NativeScript
-         |+| ScriptDataHash
-         |+| Redeemers
-         |+| GenesisHash
-         |+| GenesisDelegateHash
-         |+| AuxiliaryDataHash
-         |+| Address
-         |+| Value
-         |+| Ed25519Signature
-         |+| VRFKeyHash
-     -- Add more as needed.
-     )
-  -> ByteArray
+foreign import _toBytes
+  :: forall a. a -> ByteArray
+
+toBytes
+  :: forall a
+   . Castable a SerializationData
+  => a
+  -> CborBytes
+toBytes = CborBytes <<< _toBytes
```

Comment:
You're wrapping then unwrapping immediately at multiple call sites so why not something along the lines of? 

```suggestion
toBytes
  :: forall a
   . Castable a SerializationData
  => a
  -> CborBytes
toBytes = CborBytes <<< _toBytes

toRawBytes
  :: forall a
   . Castable a SerializationData
  => a
  -> ByteArray
toRawBytes = _toBytes
```